### PR TITLE
MCR-2684 RestAPIv2: JSON serialization for objects and derivates

### PR DESCRIPTION
**Pull request by rsteph-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2684).

### CODE REVIEW

**Review by yagee-de — COMMENTED** [review]
JSON output should be done via Transformer and is already handled via `org.mycore.restapi.converter.MCRContentAbstractWriter`. But we could include a transformer for MCRDerivate. Feel free to submit on with this PR

**Comment by yagee-de** [comment]
We could also include a Transformer for handling your case, that `MCRObject.createJSON()` is enough.

**Comment by rsteph-de** [comment]
works now with MCRContentWriter and new MCRContentTransformers to create JSON from MCRObjects and MCRDerivates

File: mycore-base/src/main/java/org/mycore/common/content/transformer/MCRBase2JSONTransformer.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+        result.setEncoding(getEncoding());
+        result.setUsingSession(source.isUsingSession());
+        return result;
+    }
+
+    private static JsonObject toJSON(MCRContent source) throws IOException {
+        try {
+            String idAttr = source.asXML().getRootElement().getAttributeValue("ID");
+            MCRObjectID mcrId = MCRObjectID.getInstance(idAttr);
+            MCRBase mcrBase = MCRMetadataManager.retrieve(mcrId);
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/common/content/transformer/MCRBase2JSONTransformer.java]: You should use the content provided not read it again.

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-base/src/main/java/org/mycore/common/content/transformer/MCRBase2JSONTransformer.java (hunk 1 above)
Comment (rsteph-de) [reply mycore-base/src/main/java/org/mycore/common/content/transformer/MCRBase2JSONTransformer.java]: solved with new ContentTransformers vor MCRObject and MCRDerivate

**Review by yagee-de — APPROVED** [review]
:+1: looks very good

**Pull request merged**